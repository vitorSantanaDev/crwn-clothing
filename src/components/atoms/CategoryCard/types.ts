export interface IViewProps {
  title: string
  action: string
  urlImageBackground: string
}
