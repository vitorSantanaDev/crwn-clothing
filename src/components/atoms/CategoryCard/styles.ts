import styled, { css } from 'styled-components'
import { IViewProps } from './types'

export const CategoryContainer = styled.div`
  ${() => css`
    height: 240px;
    flex: 1 1 auto;
    display: flex;
    min-width: 30%;
    overflow: hidden;
    align-items: center;
    margin: 0 7.5px 15px;
    justify-content: center;
    border: 1px solid black;

    &:hover {
      cursor: pointer;
      .bodyContent {
        opacity: 0.9;
      }
    }

    &.large {
      height: 380px;
    }

    &:first-child {
      margin-right: 7.5px;
    }

    &:last-child {
      margin-left: 7.5px;
    }
  `}
`
export const CategoryBackground = styled.div<Partial<IViewProps>>`
  ${({ urlImageBackground }) => css`
    width: 100%;
    height: 100%;
    background-size: cover;
    background-position: center;
    background-image: url(${urlImageBackground});

    &:hover {
      transform: scale(1.1);
      transition: transform 6s cubic-bezier(0.25, 0.45, 0.45, 0.95);
    }
  `}
`
export const CategoryBody = styled.div`
  ${() => css`
    height: 90px;
    padding: 0 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border: 1px solid black;
    background-color: white;
    opacity: 0.7;
    position: absolute;
  `}
`
export const CategoryTitle = styled.h2`
  ${() => css`
    font-weight: bold;
    margin: 0 6px 0;
    font-size: 22px;
    color: #4a4a4a;
  `}
`
export const CategoryAction = styled.p`
  ${() => css`
    font-weight: lighter;
    font-size: 16px;
  `}
`
