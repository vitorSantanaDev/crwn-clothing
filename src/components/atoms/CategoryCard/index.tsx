import React from 'react'
import { useNavigate } from 'react-router-dom'

import { IViewProps } from './types'

import * as S from './styles'

export default function CategoryCard({
  title,
  action,
  route,
  urlImageBackground
}: IViewProps) {
  const navigate = useNavigate()

  const onNavigateHandler = () => navigate(route)

  return (
    <S.CategoryContainer onClick={onNavigateHandler}>
      <S.CategoryBackground urlImageBackground={urlImageBackground} />
      <S.CategoryBody className="bodyContent">
        <S.CategoryTitle>{title}</S.CategoryTitle>
        <S.CategoryAction>{action}</S.CategoryAction>
      </S.CategoryBody>
    </S.CategoryContainer>
  )
}
