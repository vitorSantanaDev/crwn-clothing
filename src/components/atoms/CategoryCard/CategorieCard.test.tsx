import { render, screen } from '@testing-library/react'
import CategoryCard from '.'

describe('<CategoryCard />', () => {
  it('should render the component correctly', () => {
    render(
      <CategoryCard
        title="Hats"
        action="/"
        urlImageBackground="https://i.ibb.co/ypkgK0X/blue-beanie.png"
      />
    )
    const heading = screen.getByRole('heading', { name: /hats/i })
    expect(heading).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(
      <CategoryCard
        title="Hats"
        action="/"
        urlImageBackground="https://i.ibb.co/ypkgK0X/blue-beanie.png"
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
