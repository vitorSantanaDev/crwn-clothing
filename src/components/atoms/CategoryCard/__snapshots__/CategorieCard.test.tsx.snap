// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCard /> should match snapshot 1`] = `
<div
  class="sc-bczRLJ cfOrxV"
>
  <div
    class="sc-gsnTZi gyuSTl"
  />
  <div
    class="sc-dkzDqf eyHLhi bodyContent"
  >
    <h2
      class="sc-hKMtZM gvJbuC"
    >
      Hats
    </h2>
    <p
      class="sc-eCYdqJ UOQJv"
    >
      /
    </p>
  </div>
</div>
`;
